ENT-3109: Create Jenkins job for translation extraction

sonalidhome opened the pull request:
(gettext/msgattrib)
 - Added a script to execute the `gettext` and `msgattrib` tasks.
   This is gets executed for master, candlepin-3.1-HOTFIX, and candlepin-2.9-HOTFIX
   branches. After the execution, the changed files are committed and pushed in
   respective branches of candlepin.
 - Added job which runs periodically on every Sunday and Thursday 7 AM.


**Note:** PR - https://github.com/candlepin/candlepin/pull/2850 need to be merged first.

nikosmoum commented on src/jobs/candlepinInternationalizationJob.groovy:
```diff
@@ -0,0 +1,17 @@
+
+job("$baseFolder/candlepin-internationalization") {
+    description('Periodically executes the gettext and msgattrib tasks on master/hotfix branches')
+    label('rhsm')
```
One issue we need to solve: This job will need to run on a node that has both java 11 and 8, and the nodes denoted by this label don't have java 11 afaik.

@Lorquas Would it be better if we use the `candlepinbuild` label (I think this one is used in the release job that has both versions installed)?

Lorquas replied:
that should work for now

sonalidhome replied:
Updated.

nikosmoum commented on src/resources/candlepin-internationalization.sh:
```diff
@@ -0,0 +1,73 @@
+  sudo git commit -m "Internationalization: common/po file is updated"
+  evalrc $? "Git commit was not successful for branch $GIT_BRANCH."
+  #TODO: Uncomment below once the script is approved
+  #sudo git push $GIT_BRANCH
```
Not only do we need to uncomment this, but I'm not even sure if it will let you push like that. We might need to use a github token/credential in the same way we use in the candlepin_release job (see file nanny-scripts/jenkins/candlepin-release.pipeline.groovy in our nanny-script gitlab repo)

sonalidhome replied:
Agreed. Updated to push using credentials.

nikosmoum reviewed (commented):
ACK in general, but we need to think about a couple of things that need attention first

nikosmoum commented on src/resources/candlepin-internationalization.sh:
```diff
@@ -0,0 +1,73 @@
+  sudo ./gradlew msgattrib
+  evalrc $? "msgattrib was not successful for branch $GIT_BRANCH."
+
+  #Code to commit the updated files
```
One more important thing I forgot: There is a chance (in fact, I'd say that will be most times this job will run, especially for the HOTFIX branches), that there will be no actual translation strings being removed at all, and therefore there will be no commit needed.

Unfortunately, just running `./gradlew gettext`, will alter the common/po/keys.pot file by altering the `POT-Creation-Date:` in that file, even though no other changes were made. in that case, we don't want to git add/commit/push at all (no commits should be done for that branch), so we need a way to detect this at this point in the script.

The nanny-script actually does perform this check by running `git diff | egrep -v -e '^( |\+#|\-#|@@|\+\+\+|\-\-\-|diff|index)' -e 'X-Generator' -e 'POT-Creation-Date'`. If this returns nothing, then nothing significant has changed, so we shouldn't commit/push anything.

sonalidhome replied:
Updated.

Commit:
ENT-3109: Create Jenkins job for translation extraction
(gettext/msgattrib)
 - Added a script to execute the `gettext` and `msgattrib` tasks.
   This is gets executed for master, candlepin-3.1-HOTFIX, and candlepin-2.9-HOTFIX
   branches. After the execution, the changed files are committed and pushed in
   respective branches of candlepin.
 - Added job which runs periodically on every Sunday and Thursday 7 AM.

nikosmoum commented on src/resources/candlepin-internationalization.sh:
```diff
@@ -0,0 +1,83 @@
+
+  else
+    #To stash the unnecessary changes
+    sudo git stash
```
Why stash them? At this point we want to discard them entirely using `sudo git restore .`

sonalidhome replied:
I was trying out it in the local vagrant box. It has an older versioned git. This command is not recognized there. Hence used a stash to leave those changes and later this directory will get cleaned up in the cleanup job.

nikosmoum replied:
That's curious. I didn't know the restore command is kind of new. stash should be fine then
